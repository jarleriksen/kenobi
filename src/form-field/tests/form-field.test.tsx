import React from 'react';
import { render } from '@testing-library/react';
import { FormField } from '../';
import '@testing-library/jest-dom/extend-expect';

describe('Form Field', () => {
  test('it renders correct elements', () => {
    const { getByTestId } = render(
      <FormField
        label="A required text input field"
        description="Text input description"
        errorMessage="This field is required">
        <span>Form Field</span>
      </FormField>,
    );

    const formField = getByTestId('kenobi-form-field');
    const label = getByTestId('kenobi-form-field-label');
    const description = getByTestId('kenobi-form-field-description');
    const errorMessage = getByTestId('kenobi-form-field-error-message');

    expect(formField).toContainElement(label);
    expect(formField).toContainElement(description);
    expect(formField).toContainElement(errorMessage);
  });
});
