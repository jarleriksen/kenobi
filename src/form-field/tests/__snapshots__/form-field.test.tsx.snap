// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Field it renders correct elements 1`] = `
<div
  class="sc-bxivhb kwamHj"
  data-testid="kenobi-form-field"
>
  <label
    class="sc-ifAKCX dxpRIg sc-bdVaJa feWKzF"
    color="default"
    data-testid="kenobi-form-field-label"
    font-family="ui"
    for=""
    size="400"
  >
    A required text input field
  </label>
  <p
    class="sc-EHOje hJqfXO sc-bwzfXH fcopls"
    color="muted"
    data-testid="kenobi-form-field-description"
    size="400"
  >
    Text input description
  </p>
  <span>
    Form Field
  </span>
  <div
    class="sc-bZQynM fuFeTf"
    data-testid="kenobi-form-field-error-message"
  >
    <svg
      data-icon="error"
      data-testid="kenobi-icon"
      height="14"
      style="margin-right: 8px; fill: #ff3a57;"
      viewBox="0 0 16 16"
      width="14"
    >
      <path
        d="M7.99-.01c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm1 13h-2v-2h2v2zm0-3h-2v-7h2v7z"
        fill-rule="evenodd"
      />
    </svg>
    <p
      class="sc-bwzfXH gqNhMU"
      color="danger"
      role="alert"
      size="300"
    >
      This field is required
    </p>
  </div>
</div>
`;
